### PR TITLE
Update to ginkgo/v2

### PR DESCRIPTION
**Pull request by liornoy**
This PR is bumping ginkgo to v2

### CODE REVIEW

**Comment by yanirq** [comment]
@liornoy  gingko V2 is now under NTO.  you can update the vendoring of NTO to the latest  so it will be included under this PR
cc @SchSeba

File: go.mod [hunk 1]
```diff
@@ -106,47 +105,48 @@ require (
...
+	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
-	k8s.io/apiserver v0.25.0 // indirect
-	k8s.io/component-base v0.25.2 // indirect
-	k8s.io/klog/v2 v2.80.1 // indirect
-	k8s.io/kube-openapi v0.0.0-20220803162953-67bda5d908f1 // indirect
-	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	k8s.io/apiserver v0.26.1 // indirect
```
Comment (SchSeba) [inline go.mod:145]: you bump the k8s api here but there is a pinned version below please update it to 0.26.1

File: go.mod [hunk 2]
```diff
@@ -1,102 +1,101 @@
...
 	github.com/openshift/client-go v3.9.0+incompatible
 	github.com/openshift/cluster-nfd-operator v0.0.0-00010101000000-000000000000
 	github.com/openshift/cluster-node-tuning-operator v0.0.0-00010101000000-000000000000
-	github.com/openshift/machine-config-operator v0.0.1-0.20220706180257-35d79621a587
-	github.com/openshift/ptp-operator v0.0.0-20221109224215-0229788f84f5
-	github.com/spf13/cobra v1.4.0
-	github.com/stretchr/testify v1.7.1
-	golang.org/x/sys v0.0.0-20220722155257-8c9f86f7a55f
+	github.com/openshift/machine-config-operator v0.0.1-0.20230118083703-fc27a2bdaa85
+	github.com/openshift/ptp-operator v0.0.0-20230120025914-ae56875d0f51
```
Comment (SchSeba) [inline go.mod:31]: please put this one in the replace section below with all the other cnf related projects

File: go.mod [hunk 3]
```diff
@@ -187,23 +187,22 @@ replace (
 	github.com/cri-o/cri-o => github.com/cri-o/cri-o v1.18.1
 	github.com/go-log/log => github.com/go-log/log v0.1.0
 	github.com/gorilla/websocket => github.com/gorilla/websocket v1.4.2
+	github.com/k8stopologyawareschedwg/resource-topology-exporter => github.com/k8stopologyawareschedwg/resource-topology-exporter v0.8.0
 	github.com/mtrmac/gpgme => github.com/mtrmac/gpgme v0.1.1
 	github.com/openshift/api => github.com/openshift/api v0.0.0-20210713130143-be21c6cb1bea // release-4.8
 	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20210521082421-73d9475a9142 // release-4.8
 	github.com/openshift/cluster-node-tuning-operator => github.com/openshift/cluster-node-tuning-operator v0.0.0-20221109154259-940cf4ab1d74 // release-4.13
```
Comment (SchSeba) [inline go.mod:194]: did you bump this one?
please put it in the test deps section

File: cnf-tests/Dockerfile [hunk 4]
```diff
@@ -1,12 +1,12 @@
-FROM registry.ci.openshift.org/openshift/release:golang-1.17 AS builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.18 AS builder
 WORKDIR /go/src/github.com/openshift-kni/cnf-features-deploy
 COPY . .
 RUN make test-bin
 RUN git rev-list -1 HEAD > ./cnf-tests/bin/cnftests-sha.txt
 
 FROM quay.io/openshift/origin-oc-rpms:4.8 AS oc
```
Comment (yanirq) [inline cnf-tests/Dockerfile:7]: is'nt 4.8 here a bit outdated ?

File: go.mod (hunk 1 above)
Comment (liornoy) [reply go.mod:145]: done, pinned k8s to 0.26.1

File: go.mod (hunk 2 above)
Comment (liornoy) [reply go.mod]: done

File: cnf-tests/Dockerfile (hunk 4 above)
Comment (liornoy) [reply cnf-tests/Dockerfile:7]: yep, will update. thanks.

File: go.mod (hunk 3 above)
Comment (liornoy) [reply go.mod]: done

**Comment by yanirq** [comment]
I suggest running `make check-tests-nodesc `locally to clean and adjust the tests requirements first

**Comment by liornoy** [comment]
> I suggest running `make check-tests-nodesc` locally to clean and adjust the tests requirements first

I ran it, plus added manually test descriptions from the metallb repo.
Regarding the empty testcases on the top of the file, it's an issue with ginkgo's junit reporter new behavior.
I filed an issue on the ginkgo repo to suggest a solution for that: https://github.com/onsi/ginkgo/issues/1145

Please let me know wdyt. :)
@yanirq

**Comment by yanirq** [comment]
> > I suggest running `make check-tests-nodesc` locally to clean and adjust the tests requirements first
> 
> I ran it, plus added manually test descriptions from the metallb repo. Regarding the empty testcases on the top of the file, it's an issue with ginkgo's junit reporter new behavior. I filed an issue on the ginkgo repo to suggest a solution for that: [onsi/ginkgo#1145](https://github.com/onsi/ginkgo/issues/1145)
> 
> Please let me know wdyt. :) @yanirq

If its not blocking CI we should be good and in the future we can bump the ginkgo minor version including your fix and align

**Comment by yanirq** [comment]
2 known failures are failing on e2e-gcp-ovn  and will be addressed in other PRs. 
```
[FAIL] [rfe_id:27368][performance] RPS configuration [It] [test_id: 59572] Check RPS Mask is applied to atleast one single rx queue on all veth interface

 /go/src/github.com/openshift-kni/cnf-features-deploy/vendor/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/performance.go:323

 [FAIL] [tuningcni] sysctl allowlist update [It] should start a pod with custom sysctl only after adding sysctl to allowlist

 /go/src/github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/e2esuite/security/tuning.go:186
```

File: cnf-tests/Dockerfile [hunk 5]
```diff
@@ -4,7 +4,7 @@ COPY . .
 RUN make test-bin
 RUN git rev-list -1 HEAD > ./cnf-tests/bin/cnftests-sha.txt
 
-FROM quay.io/openshift/origin-oc-rpms:4.8 AS oc
+FROM quay.io/openshift/origin-oc-rpms:4.13 AS oc
```
Comment (ffromani) [inline cnf-tests/Dockerfile:7]: for the next time: bumping these images is best done in separate PR (or we should update our prow config to consume the current PR, which is probably a bigger undertaking)

File: ztp/resource-generator/Containerfile [hunk 6]
```diff
@@ -1,5 +1,5 @@
 # Builder
-FROM registry.ci.openshift.org/openshift/release:golang-1.17 as builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.19 as builder
```
Comment (ffromani) [inline ztp/resource-generator/Containerfile:2]: ditto, these bumps are usually better done on their own PR

File: cnf-tests/TESTLIST.md [hunk 7]
```diff
@@ -9,6 +9,7 @@ hack/fill-empty-docs.sh, check the json description files and fill the missing d
 The validation tests are preliminary tests intended to verify that the instrumented features are available on the cluster.
 | Test Name | Description |
 | -- | ----------- |
+|  | XXXXXXXX |
```
Comment (ffromani) [inline cnf-tests/TESTLIST.md:12]: this doesn't look right (there are other instances below)

**Review by ffromani — COMMENTED** [review]
let's doublecheck `TESTLIST.md`

File: cnf-tests/TESTLIST.md (hunk 7 above)
Comment (liornoy) [reply cnf-tests/TESTLIST.md:12]: This is caused by the new behavior of ginkgo's JUnit reporter, I contributed a fix already to ginkgo and it is going to be released around next week, I'll update the ginkgo's minor version then.

Comment (ffromani) [reply cnf-tests/TESTLIST.md:12]: ok, I guess we can move on then.

Comment (liornoy) [reply cnf-tests/TESTLIST.md:12]: oh
I see now it's already out

Comment (ffromani) [reply cnf-tests/TESTLIST.md:12]: cool, so let's prepare a followup PR to bump the dep and clean up this leftover
